docs/Practices/hub/hub.md

```diff
@@ -0,0 +1,118 @@
+
+- [UA: Errors as Values](./learning.md#errors-as-values), [UA: Domain Driven Design - Domain Modeling](./learning.md#domain-modeling) The known errors of a command/query within a Vertical Slice should be included in its type signature. *It is recommended to coerce exceptions to values for known errors whenever possible, but at minimum no known errors should be exposed from a command/query function by way of thrown exception.*
+
+- [UA: Domain Driven Design - Domain Modeling](./learning.md#domain-modeling) If an error represents a possible bad data state there should be documentation regarding this bad data state and how to remediate it. *At minimum there should be a comment attached to the type definition for a given error that provides appropriate context.*
+
+- [UA: Testing](./learning.md#testing) When a bug is fixed an existing test should be updated to "prove" the fix or a new test should be written to accompany the fix.
+
+- [UA: Validation](./learning.md#validatation) Validation should be performed on all handler request data.
+
+## Remediation of code not meeting guidelines
+There are cases of existing code not meeting these existing guidelines. Refactoring existing code to meet these guidelines should be seen as tech debt and prioritized accordingly. Effort should be taken to have all new code meet these guidelines. 
+
+Any code that must break a guideline should be called out, addressed, and either documented for cases which make sense to break a guideline, or have an associated tech debt ticket created in cases where breaking a guideline is required to quickly remediate a critical issue that following a guideline would delay. *An example of this might be a critical bug that needs to be out ASAP. Maybe a fix was written and through existing automated tests + some manual testing + developer intuition it is determined that this fix works and won't make the problem worse. In this case it is pragmatic to get this fix out even if it ignores the guideline for testing to prove a fix.*
+
+## Conventions
+Sometimes guidance that provides more specificity than a guideline needs to exist. 
+
+
+More often than not this is for enabling better implication. For example a naming convention allows engineers to draw better and more accurate conclusions without needing to review documentation.
+
+Additionally guidelines might be library, framework, or project specific. Defining rules around how to interact with a library for example might be pragmatic if that library has capabilities that by some usages might conflict with existing guidelines.
+
+For the purpose of this document, conventions should be seen as more specific guidelines or extensions to the guidelines.
+
+Naming:
+- The entrypoint function for a command or query should start with `command` or `query`.
```

Comment:
What about ending a function name with `command` or `query`?

Reply:
I'm indifferent to prefix or postfix. Would you be amenable to a followup proposal PR that postfix by amending the rule in this document + another references PR that modifies the function names in Hub?

Reply:
Sure.